packages/api/src/services/actions.ts

```diff
@@ -47,48 +47,47 @@ export function Handlers(config: Config, appState: Dependencies): Actions {
       assert(exists(prices[currency]), "invalid currency type: " + currency);
       return prices[currency].latest;
     },
-    async latestPriceByAddress(address: string, currency: CurrencySymbol = "usd") {
-      assert(address, "requires an erc20 token address");
-      assert(exists(prices[currency]), "invalid currency type: " + currency);
-      const priceSample = prices[currency].latest[address];
-      assert(exists(priceSample), "No price for address: " + address);
-      return priceSample;
-    },
-    // synthetic prices are determined by bot pricefeeds
-    async latestSynthPriceByAddress(address: string) {
-      assert(address, "requires an erc20 token address");
-      const priceSample = synthPrices.latest[address];
-      assert(exists(priceSample), "No synthetic price for address: " + address);
-      return priceSample;
-    },
-    async allLatestSynthPrices() {
-      return synthPrices.latest;
-    },
-    async historicalPricesByAddress(
-      address: string,
-      start = 0,
-      end: number = nowS(),
-      currency: "usd" = "usd"
-    ): Promise<PriceSample[]> {
-      assert(start >= 0, "requires a start value >= 0");
-      assert(exists(prices[currency]), "invalid currency type: " + currency);
-      assert(exists(prices[currency].history[address]), "no prices for address" + address);
-      const results = await prices[currency].history[address].betweenByTimestamp(start, end);
-      // convert this to tuple to save bytes.
-      return results.map(({ price, timestamp }) => [timestamp, price]);
-    },
-    async sliceHistoricalPricesByAddress(
-      address: string,
-      start = 0,
-      length = 1,
-      currency: "usd" = "usd"
-    ): Promise<PriceSample[]> {
-      assert(start >= 0, "requires a start value >= 0");
-      assert(exists(prices[currency]), "invalid currency type: " + currency);
-      assert(exists(prices[currency].history[address]), "no prices for address" + address);
-      const results = await prices[currency].history[address].sliceByTimestamp(start, length);
-      // convert this to tuple to save bytes.
-      return results.map(({ price, timestamp }) => [timestamp, price]);
+    // get prices by token address
+    latestPriceByTokenAddress: queries.latestPriceByTokenAddress,
```

Comment:
these calls were refactored to be more clear on specifying token address vs emp address when getting a price. The calls essentially allow someone to query latest price or price  history by either emp (collateral or synthetic) or the actual erc20 address of the token

Reply:
Its a bit confusing that the function name is `latestSyntheticPrice` but you can query the EMP's collateral price?

Reply:
Nice, this name is much clearer